qa/ingest-with-mustache/src/test/java/org/elasticsearch/ingest/IngestDocumentMustacheIT.java

```diff
@@ -44,6 +44,30 @@ public void testAccessMetaDataViaTemplate() {
         assertThat(ingestDocument.getFieldValue("field1", String.class), equalTo("2 bar"));
     }
 
+    // TODO(talevy): expected?
+    public void testAccessSubFieldsViaTemplate() {
+        Map<String, Object> document = new HashMap<>();
+        Map<String, Object> fooDoc = new HashMap<>();
+        fooDoc.put("bar", "value");
+        document.put("foo", fooDoc);
+        IngestDocument ingestDocument = new IngestDocument("index", "type", "id", null, null, null, null, document);
+        ingestDocument.setFieldValue(templateService.compile("field1"), ValueSource.wrap("1 {{foo.bar}}", templateService));
+        assertThat(ingestDocument.getFieldValue("field1", String.class), equalTo("1 value"));
+    }
+
+    // TODO(talevy): expected?
+    public void testAccessFieldWithDotsViaTemplate() {
+        Map<String, Object> document = new HashMap<>();
+        Map<String, Object> fooBarDoc = new HashMap<>();
+        fooBarDoc.put("val", "value");
+        document.put("foo.bar", fooBarDoc);
+        IngestDocument ingestDocument = new IngestDocument("index", "type", "id", null, null, null, null, document);
+        ingestDocument.setFieldValue(templateService.compile("field1"), ValueSource.wrap("1 {{foo.bar}}", templateService));
+        assertThat(ingestDocument.getFieldValue("field1", String.class), equalTo("1 {val=value}"));
+        ingestDocument.setFieldValue(templateService.compile("field2"), ValueSource.wrap("1 {{foo.bar.val}}", templateService));
+        assertThat(ingestDocument.getFieldValue("field2", String.class), equalTo("1 "));
+    }
+
```

Comment:
sounds good. thought I would leave it in for another pass so we can discuss it in context. I will remove it from this PR